### PR TITLE
Updated clients with additional pre-existing functionality

### PR DESCRIPTION
**Pull request by pittenga**
*Issue #, if available:*

*Description of changes:*
Based on our usage of CDF, the following updates were made to make some of the clients function as desired (and have comparable functionality to the APIs)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by boardthatpowder** [comment]
Hi @pittenga. Thanks for the PR. It's failing to build. Can you take a look, then validate with `rush build` and `rush test`. Thanks.

```
==[ FAILURE: 2 projects ]======================================================

--[ FAILURE: @cdf/assetlibrary-client ]-------------[ 1 minute 15.9 seconds ]--

Invoking: npx tsc -b

src/client/templates.lambda.service.ts(18,81): error TS2305: Module '"@cdf/lambda-invoke"' has no exported member 'QueryStringParametersDictionary'.

--[ FAILURE: @cdf/assetlibrary-history ]------------[ 1 minute 26.1 seconds ]--

Invoking: npx tsc -b && npx shx cp -R 'src/config' 'dist/'

../../libraries/clients/assetlibrary-client/src/client/templates.lambda.service.ts(18,81): error TS2305: Module '"@cdf/lambda-invoke"' has no exported member 'QueryStringParametersDictionary'.

Projects failed to build.
```

**Comment by aaronatbissell** [comment]
Done! Sorry about that!

File: source/packages/libraries/clients/assetlibrary-client/src/client/templates.apigw.service.ts [hunk 1]
```diff
@@ -64,7 +64,12 @@ export class TemplatesApigwService extends TemplatesServiceBase implements Templ
         ow(resource.templateId,'templateId', ow.string.nonEmpty);
         ow(resource.category,'category', ow.string.nonEmpty);
 
-        const url = `${this.baseUrl}${super.templateRelativeUrl(resource.category, resource.templateId)}`;
```
Comment (bahavar) [inline source/packages/libraries/clients/assetlibrary-client/src/client/templates.apigw.service.ts:67]: What is the reasoning behind this change ?!  I dont see any functionality change

File: source/packages/libraries/clients/assetlibrary-client/src/client/templates.lambda.service.ts [hunk 2]
```diff
@@ -68,8 +68,13 @@ export class TemplatesLambdaService extends TemplatesServiceBase implements Temp
         ow(resource.templateId,'templateId', ow.string.nonEmpty);
         ow(resource.category,'category', ow.string.nonEmpty);
 
+        const templateId = resource.templateId;
```
Comment (bahavar) [inline source/packages/libraries/clients/assetlibrary-client/src/client/templates.lambda.service.ts:71]: What is the reasoning behind this change ?!

Comment (aaronatbissell) [reply source/packages/libraries/clients/assetlibrary-client/src/client/templates.lambda.service.ts:71]: You're right - no real functionality change here, but it is consistent with the `createTemplate` functionality. When developing, it was a little confusing when creating/updating templates to see one delete your `resource.templateId`/`resource.category`, but the other one did not.

File: source/packages/libraries/clients/assetlibrary-client/src/client/templates.apigw.service.ts (hunk 1 above)
Comment (aaronatbissell) [reply source/packages/libraries/clients/assetlibrary-client/src/client/templates.apigw.service.ts:67]: Same comment - no real functionality change here, but it is consistent with the `createTemplate` functionality. When developing, it was a little confusing when creating/updating templates to see one delete your `resource.templateId`/`resource.category`, but the other one did not.

**Review by bahavar — APPROVED** [review]

**Pull request merged**